### PR TITLE
Practice 5: todo list

### PR DESCRIPTION
**Pull request by awantika-b20150**
### Problem
https://github.com/pj100-team/react-v3/issues/5

### Changes made
Made form using react-hook-form, with custom error messages and additional search button that searches for address using the api. And if the address is found, display in console and fill rest of the fields else display error address not found.

### Demo Video

https://github.com/user-attachments/assets/e583674a-64e9-4346-9fd4-7595e6e243d9

### CODE REVIEW

File: src/templates/Practice5.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  id: number;
+  todayDate: string;
+  taskName: string;
+  isChecked: boolean;
+}
+
+const Practice5: React.FC = () => {
+  const [todos, setTodos] = useState<Data[]>([]);
+  const [task, setTask] = useState<string>('');
+  const [allSelected, areAllSelected] = useState<boolean>(false);
```
Comment (ThanainanLi) [inline src/templates/Practice5.tsx]: To Follow our [React guidelines ](https://pj100.esa.io/posts/16722#%E3%82%A2%E3%83%B3%E3%83%81%E3%83%91%E3%82%BF%E3%83%BC%E3%83%B3%E3%81%AE%E6%94%B9%E5%96%84)

```suggestion
  const [isAllSelected, setIsAllSelected] = useState<boolean>(false);
```

File: src/templates/Practice5.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  const selectAll = () => {
+    const newAllSelected = !allSelected;
+    areAllSelected(newAllSelected);
+    setTodos(todos.map(todo => ({ ...todo, isChecked: newAllSelected })));
+  };
```
Comment (ThanainanLi) [inline src/templates/Practice5.tsx]: Should we go to this way to toggle the state? 

```suggestion
  const selectAll = () => {
  setIsAllSelected(prevIsAllSelected => {
    const newState = !prevIsAllSelected;
    setTodos(todos => todos.map(todo => ({ ...todo, isChecked: newState })));
    return newState;
  });
};
```

**Review by ThanainanLi — COMMENTED** [review]
Thank you for your work! Few comments from me!

File: src/templates/Practice5.tsx (hunk 1 above)
Comment (awantika-b20150) [reply src/templates/Practice5.tsx]: Made the changes.

File: src/templates/Practice5.tsx (hunk 2 above)
Comment (awantika-b20150) [reply src/templates/Practice5.tsx]: Yes we can go this way but are you suggesting it in terms of cleaner way and better readability or is this way more efficient too ?

Comment (ThanainanLi) [reply src/templates/Practice5.tsx]: @awantika-b20150 For better readability and also make sure that the state is completely sync. What do you think?

Comment (awantika-b20150) [reply src/templates/Practice5.tsx]: Understood. I think this is better for syncing the state of our individual toggles and select all toggle. I have made the changes.

File: src/templates/Practice5.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+
+
+interface Data {
```
Comment (extramix) [inline src/templates/Practice5.tsx]: Could you change this name into something more specific like `Todo` or `Task`?

`Data` doesn't say much about what's inside 🧐

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  children,
+  className = '',
+  type = 'button',
+  onClick,
+  title,
+}: ButtonProps) => {
+  return (
+    <button
+      onClick={onClick}
+      className={`text-gray-50 px-16 py-2 rounded-lg font-medium border-2 border-transparent 'bg-gray-500 hover:bg-gray-70 rounded'`}
```
Comment (extramix) [inline src/components/Button.tsx]: There is the extra `'` at the end of `rounded` classname so please remove it!

**Review by extramix — CHANGES_REQUESTED** [review]
Just a bit more from me!

File: src/templates/Practice5.tsx (hunk 3 above)
Comment (awantika-b20150) [reply src/templates/Practice5.tsx]: Made the changes

File: src/components/Button.tsx (hunk 4 above)
Comment (awantika-b20150) [reply src/components/Button.tsx]: Made the changes

**Review by extramix — APPROVED** [review]

**Review by ThanainanLi — APPROVED** [review]

**Pull request merged**